rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Kubernetes admission control threat model                           |
+| Start Date   | 04/18/2022                          |
+| Category     | Security                       |
+| RFC PR       | https://github.com/kubewarden/rfc/pull/2  |
+
+# Summary
+[summary]: #summary
+
+Describe the threat model for the admissions control, how it can effect Kubewarden
+and the possible mitigations.
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this is turn Kubewarden mode secure to operate.
+
+## Examples / User Stories
+[examples]: #examples
+
+As Kubewarden developer, I want to know which are the weak point of the Kubernetes admission control feature, so I can
+mitigate this problem.
+
+As Kubewarden user, I want to know which are the weak points of the Kubernetes admission control, so I can apply the mitigation
+in my cluster.
+
+# Detailed design
+[design]: #detailed-design
+
+The threat model of the Kubernetes admission control list 18 possible threats.
+All of them will be discussed sharing a brief description of it, possible mitigation
+and what we, as Kubewarden team, should do about it (if necessary). As well as
+help our users to be aware of the risks.
+
+If more details about each threat is needed, go to the original document here:
+https://github.com/kubernetes/sig-security/tree/main/sig-security-docs/papers/admission-control
+
+
+## Threat 1 - Attacker floods webhook with traffic preventing its
+
+An attacker who has access to the Webhook endpoint, at the network level, could
+send large  quantities of traffic, causing an effective denial of service to the
+admission controller.
+
+**Mitigation**
+
+Webhook fails closed. In other words, if the webhook does not respond in time,
+for any reasion, API server should reject the request. We are safe on this.
+Kubewarden default behavior  already does that.
```

Comment:
Given the policy-server service is reachable from within the cluster, it would be possible to attempt a denial of service by accessing the service from within the cluster, by requesting a big amount of requests.

If the policy is configured to fail open (`failurePolicy: ignore`), then policies will have no effects if the endpoint is not reachable. If the policy is configured to fail closed (`failuePolicy: fail`), then it can cause a denial of service if the webhooks are not reachable.

Network policies would mitigate this threat.

Reply:
I think the only reasonable way to mitigate this issue is to introduce allow each policy to run for at most X seconds. Once the timeout is reached the evaluation response will be a rejection accompanied by a certain error code and message.